Hyeongseok

kuk6933 opened the pull request:
Week1 Assignment

gen-com committed:
Add Week 1

gen-com committed:
Update README.md

gen-com committed:
Update README.md

kuk6933 committed:
Add Title "Memorize!"

kuk6933 committed:
Make plus, minus function

kuk6933 committed:
Create theme function and change structure

kuk6933 committed:
@ObservedObject -> @EnvironmentObject

kuk6933 committed:
plus, minus button updapte

kuk6933 committed:
FooterView Update

gen-com commented on week1/Memorize/Memorize/ContentView.swift:
```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+struct ContentView: View {
+
```
선언 후 공백을 둘 지 두지 않을 지 통일해주세요.

kuk6933 replied:
넵

kuk6933 replied:
리뷰 감사합니다!

gen-com commented on week1/Memorize/Memorize/ContentView.swift:
```diff
@@ -0,0 +1,36 @@
+
+struct ContentView: View {
+
+    @StateObject private var toBeObserved = ToBeObserved()
```
@StateObject를 사용한 이유는 무엇인가요 ?

@ObservedObject와는 어떤 차이가 있나요 ?

kuk6933 replied:
StateObject는 하나의 객체로 만들어지고 VIew가 얼마나 초기화되든지 상관없이 별개의 객체로 관리되어서 ObservableObect에서 변화가 일어나면 @ObservedObject는 View를 처음부터 다시 그리는 것가 달리 @StateOnject는 ObservableObject의 데이터가 들어간 부분만 View를 다시그린다고 알고있습니다. 
그래서 애플에서 ObservableObject를 처음 초기화할 때는 StateObject를 사용하고 이후에 이를 넘겨받을 때는 ObservedObject를 사용하는것을 추천한다고 해서 @StateObject를 사용했습니다.

gen-com replied:
@StateObject는 @ObservedObject와 유사합니다. 다만 단 하나의 원본임을 뜻합니다.

@StateObject private var toBeObserved = ToBeObserved() (O)
@ObservedObject private var toBeObserved = ToBeObserved() (X)

변화에 대해서 차이가 있는 것은 어디서 봤나요 ? 자료를 부탁드립니다.

ContentView가 스크린에서 사라지면 해당 변수는 어떻게 될까요 ?

kuk6933 replied:
https://wondev.tistory.com/5 
이 블로그를 참고했습니다.
contentView가 사라지만 다른 View에서 EnvironmnObject가 작동하지 못할거라 생각합니다.

gen-com replied:
https://developer.apple.com/documentation/swiftui/managing-model-data-in-your-app
해당 애플 문서에서 @StateObject와 @ObservedObject 차이점을 다루는 부분을 읽어보고, 가져온 블로그의 해석에 대해 다시 생각해 볼까요 ?

kuk6933 replied:
네 알겠습니다!

gen-com commented on week1/Memorize/Memorize/CardListView.swift:
```diff
@@ -0,0 +1,25 @@
+import SwiftUI
+
+struct CardListView: View {
+    @EnvironmentObject var toBeObserved : ToBeObserved
```
@EnvironmentObject를 사용한 이유는 무엇인가요 ? 그리고 해당 뷰에서 선언한 이유는 무엇인가요 ?

@ObservedObject와는 어떤 차이가 있나요 ?

":" 에 대해 들여쓰기를 통일해주세요.
ex) toBeObserved: ToBeObserved

kuk6933 replied:
@EnvironmentObject는 어떤 data를 모든 VIew에서 사용해야할 때 쓰면 좋다고 알고있습니다. 해당 View에서는 display할 cardList를 받아오기 위해 사용했습니다.
@ObservedObject에서 @EnvironmentObject로 바꾼 이유는 EnvironmentObject는 Argument로 안넣어줘도 작동해서 가독성을 위해 바꿨습니다.

gen-com replied:
모든 View 확실한가요 ?

View 당 얼마나 가질 수 있나요 ?

kuk6933 replied:
모든 View까진 아니어도 대다수의 뷰에서 사용할때 쓰면 좋다고생각합니다.

kuk6933 replied:
각 View마다 @EnvironmentObject 개수에 대해 
swiftui environmentobject 
swiftui environmentobject 제한 
요런식으로 검색해봤는데 View당 얼마나 가질 수 있는지는 찾지못했습니다. 제가 생각하기에는 @Published 래퍼로 표시된 변수 개수만큼 가질 수 있지 않을까생각됩니다.

gen-com replied:
각 View가 가지는 ObservableObject 당 하나씩 가질 수 있습니다.
그렇다면 ObservableObject에 대해 여러개의 EnvironmentObject를 가지면 어떤 문제가 발생할까요 ?

kuk6933 replied:
어차피 같은 ObservableObject에 대한것이라면 메모리 낭비될거라 생각합니다.

gen-com commented on week1/Memorize/Memorize/CardListView.swift:
```diff
@@ -0,0 +1,25 @@
+                ForEach(toBeObserved.returnCardList(), id: \.self, content: { emoji in
+                    CardView(content: emoji).aspectRatio(2 / 3, contentMode: .fit)
+                })
+                
```
무의미한 공백은 지워주세요.

kuk6933 replied:
넵

gen-com commented on week1/Memorize/Memorize/FooterVIew.swift:
```diff
@@ -0,0 +1,41 @@
+
+import SwiftUI
+
+struct FooterVIew: View {
```
FooterV"I"ew -> FooterV"i"ew, 네이밍이 더 구체적이였으면 좋겠습니다.

역시 아래 공백을 통일해 주세요.

kuk6933 replied:
네

gen-com commented on week1/Memorize/Memorize/FooterVIew.swift:
```diff
@@ -0,0 +1,41 @@
+
+struct FooterVIew: View {
+
+    @EnvironmentObject var toBeObserved : ToBeObserved
```
@EnvironmentObject ?

콜론 통일

kuk6933 replied:
넵

gen-com commented on week1/Memorize/Memorize/FooterVIew.swift:
```diff
@@ -0,0 +1,41 @@
+        }
+        .font(.largeTitle)
+        .padding()
+        
```
무의미한 공백

kuk6933 replied:
수정했습니다!

gen-com commented on week1/Memorize/Memorize/FooterVIew.swift:
```diff
@@ -0,0 +1,41 @@
+
+    @EnvironmentObject var toBeObserved : ToBeObserved
+    var body: some View {
+        HStack{
```
중괄호 역시 통일해 주세요.

HStack{ -> HStack {

kuk6933 replied:
넵

gen-com commented on week1/Memorize/Memorize/CalculateButtonView.swift:
```diff
@@ -0,0 +1,45 @@
+
+import SwiftUI
+
+struct CalculateButtonView: View {
```
Button과 View를 같이 쓰는 것은 중복입니다. Button이 View에 포함되니까요 !

Calculate, 계산을 하는 버튼인 것인데 현재의 네이밍으로 카드의 수를 변경하는 것을 유추할 수 있을까요 ?

kuk6933 replied:
좀 더 직관적인 네이밍으로 수정했습니다!

gen-com commented on week1/Memorize/Memorize/CalculateButtonView.swift:
```diff
@@ -0,0 +1,45 @@
+import SwiftUI
+
+struct CalculateButtonView: View {
+    @EnvironmentObject var toBeObserved : ToBeObserved
```
@EnvironmentObject ?

콜론 통일

kuk6933 replied:
넵

gen-com commented on week1/Memorize/Memorize/CalculateButtonView.swift:
```diff
@@ -0,0 +1,45 @@
+
+struct CalculateButtonView: View {
+    @EnvironmentObject var toBeObserved : ToBeObserved
+    var isPlus: Bool
```
operation을 나누는 것은 좋습니다. Bool이 아닌 Enum을 활용해보는 것은 어떤가요 ?

kuk6933 replied:
수정해보겠습니다. 
Enum으로 바꾼다면 굉장히 짧은 코드임에도 Model에 따로 파일을 만들어서 하는게 더 좋을까요?

gen-com replied:
enum plusOrMinus 여기서 이미 구현한거 아닌가요 ? 네이밍은 좀 바꿀 필요가 있어 보이네요.

모델은 어떻게 할지도 다시 고민해 봤으면 좋겠어요.

kuk6933 replied:
review보고 Model에 수정해서 커밋한게 반영됐습니다
네이밍도 다시 바꿔봤습니다..
앱 구조에대해 고민해보겠습니다!

gen-com commented on week1/Memorize/Memorize/CalculateButtonView.swift:
```diff
@@ -0,0 +1,45 @@
+struct CalculateButtonView: View {
+    @EnvironmentObject var toBeObserved : ToBeObserved
+    var isPlus: Bool
+    init(isPlus: Bool)
```
init() {

}
위처럼 사용할 지
init()
{

}
아래처럼 사용할 지 통일해 주세요. struct, class, if 등 모두

kuk6933 replied:
넵 통일에 유의하겠습니다

gen-com commented on week1/Memorize/Memorize/CardView.swift:
```diff
@@ -0,0 +1,38 @@
+import SwiftUI
+
+struct CardView: View {
+    @State var isFaceUp = true
```
들여쓰기가 제대로 되어있지 않은 것 같습니다.

kuk6933 replied:
수정했습니다!

gen-com commented on week1/Memorize/Memorize/CardView.swift:
```diff
@@ -0,0 +1,38 @@
+               isFaceUp.toggle()
+           }
+       }
+    
```
무의미한 공백

kuk6933 replied:
수정했습니다!

gen-com commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+
+import SwiftUI
+
+enum Theme: Comparable{
```
중괄호 주의

enum을 활용한 이유는 무엇인가요 ?

kuk6933 replied:
테마가 한정되어있어 벗어나는 case는 있으면 안되기에 enum을 활용했습니다.

gen-com replied:
struct와 enum은 어떤 차이가 있나요 ?

kuk6933 replied:
struct와 달리 enum은 저장프로퍼티를 갖지 못합니다!

gen-com commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+            return Image(systemName: "face.smiling")
+        case .fruit:
+            return Image(systemName: "applelogo")
+            
```
무의미한 공백

kuk6933 replied:
넵

gen-com commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+//  Created by ohhyeongseok on 2022/07/26.
+//
+
+import SwiftUI
```
SwiftUI를 임포트한 이유는 ?

kuk6933 replied:
symbols를 Image 타입으로 사용하기 위해 임포트했습니다!

gen-com replied:
Model은 UI로부터 독립적이여야 하지 않나요 ?

kuk6933 replied:
앗.. 구조에대해 공부하고 고민해보겠습니다

gen-com commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+//
+
+import Foundation
+import SwiftUI
```
import SwiftUI, 이유는 ?

kuk6933 replied:
쓰일 거라 생각하고 임포트하고 시작했던 것 같습니다.
없어도 에러가 나지않아 삭제했습니다!

gen-com commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+import Foundation
+import SwiftUI
+
+class ToBeObserved : ObservableObject {
```
콜론 통일 !

ObservableObject로 인해 이미 관찰될 것을 알 고 있습니다. 어떤 정보를 가지고 있는 지 유추할 수 있는 네이밍이 되어야 합니다.

kuk6933 replied:
유의하겠습니다

kuk6933 replied:
이것저것 모여있어서 네이밍에 어려움을 겪었습니다. 이러한 경우에는 하나하나 파일을 분리하는게 맞을까요?

gen-com replied:
모두를 포함하는 네이밍을 하는 것이 좋습니다. MemorizeGame 과 같은.

kuk6933 replied:
넵 수정했습니다!

gen-com commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+import SwiftUI
+
+class ToBeObserved : ObservableObject {
+    @Published var cardNumber = Theme.emotion.emojis.count
```
@Published를 사용한 이유는 무엇인가요 ?

kuk6933 replied:
plus, minus버튼을 눌러 카드 개수가 변화할때마다 카드를 추가/제거한 뷰를 보여줘야 하므로 Published로 선언하여 변화를 감지할 수 있도록 했습니다.

gen-com replied:
Published는 변화를 감지하는 것인가요 ? 변화를 알리는 것인가요 ?

kuk6933 replied:
변화를 알리는 것입니다

gen-com commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+    @Published var cardNumber = Theme.emotion.emojis.count
+    @Published var currentTheme = Theme.emotion
+    
+    var themeEmojis : [String] = Theme.emotion.emojis
```
콜론

kuk6933 replied:
수정했습니다!

gen-com commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+    
+    var themeEmojis : [String] = Theme.emotion.emojis
+    
+    func changeTheme(newTheme: Theme){
```
함수 중괄호 통일 ! 아래도 마찬가지

leejun6694 replied:
`parameter` 로 `newTheme` 이름도 좋지만, `전치사`를 넣으면 더 부드러워질 것 같네요~

kuk6933 replied:
넵 수정했습니다!

gen-com commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+        themeEmojis.shuffle()
+    }
+    
+    func returnCardList() -> [String]
```
-> [String] 를 통해 어떤 값이 반환되는 지 이미 압니다. return을 네이밍에 더하지는 않았으면 합니다.

Card 리스트가 아닌 emoji들을 반환하는 것이 아닌가요 ?

func emojis -> [String] 혹은 func emojiList -> [String]

kuk6933 replied:
넵 수정했습니다! 네이밍에 더 신경쓰겠습니다

gen-com commented on week1/Memorize/Memorize/Thema.swift:
```diff
@@ -0,0 +1,8 @@
+//
+//  Created by ohhyeongseok on 2022/07/26.
+//
+
```
불필요한 파일은 지워주세요.

kuk6933 replied:
넵 모두 수정했습니다

gen-com commented on week1/Memorize/Memorize/Theme.swift:
```diff
@@ -0,0 +1,20 @@
+//
+//  Created by ohhyeongseok on 2022/07/26.
+//
+
```
어떤 역할을 하는 View 인가요 ?

kuk6933 replied:
파일 생성시 생기는 주석은 삭제했습니다!

gen-com commented on week1/Memorize/Memorize/ThemeButtonView.swift:
```diff
@@ -0,0 +1,36 @@
+//  Created by ohhyeongseok on 2022/07/27.
+//
+
+import Foundation
```
import Foundation ?

kuk6933 replied:
파일 생성시 생겼던것 같습니다. 안쓰는 import들은 삭제하겠습니다!

gen-com commented on week1/Memorize/Memorize/ThemeButtonView.swift:
```diff
@@ -0,0 +1,36 @@
+import Foundation
+import SwiftUI
+
+struct themeButtonView: View {
```
struct 이름은 대문자로 시작해야 합니다.

kuk6933 replied:
수정했습니다! 더 신경쓰겠습니다

gen-com commented on week1/Memorize/Memorize/ThemeButtonView.swift:
```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+struct themeButtonView: View {
+    @EnvironmentObject var toBeObserved : ToBeObserved
```
@EnvironmentObject ?

콜론 주의

kuk6933 replied:
수정했습니다!

gen-com commented on week1/Memorize/Memorize/ThemeButtonView.swift:
```diff
@@ -0,0 +1,36 @@
+    var theme: Theme
+    var themeName: String
+    
+    init(theme: Theme, themeName: String)
```
중괄호 통일

kuk6933 replied:
넵 수정했습니다

gen-com commented on week1/Memorize/Memorize/ThemeButtonView.swift:
```diff
@@ -0,0 +1,36 @@
+struct themeButtonView: View {
+    @EnvironmentObject var toBeObserved : ToBeObserved
+    var theme: Theme
+    var themeName: String
```
테마 이름은 Theme에 속해있도록 하는 건 어떤가요 ?

kuk6933 replied:
rawValue를 이용해 수정했습니다!

gen-com commented on week1/Memorize/Memorize/ThemeButtonView.swift:
```diff
@@ -0,0 +1,36 @@
+            }
+            .font(.system(size: 20))
+        }
+        
```
무의미한 공백

kuk6933 replied:
수정했습니다!

rose6649 commented:
commit 메세지 작성도 중요하지만,
PR에
자신이 한 일을 정리하고, 궁금한점이나 잘 안되었던것, 해결한점 등을 같이 기입하면 자신에게 더 좋은 PR이 될 것 같아요! ㅎㅎ 🥺

rose6649 commented on week1/Memorize/Memorize/FooterVIew.swift:
```diff
@@ -0,0 +1,41 @@
+    @EnvironmentObject var toBeObserved : ToBeObserved
+    var body: some View {
+        HStack{
+            CalculateButtonView(isPlus: false)
+                .padding()
+            Spacer()
+            themeButtonView(theme: .animal, themeName: "animal")
+                .padding()
+            Spacer()
+            themeButtonView(theme: .emotion, themeName: "emotion")
+                .padding()
+            Spacer()
+            themeButtonView(theme: .fruit, themeName: "fruit")
+                .padding()
+            Spacer()
+            CalculateButtonView(isPlus: true)
+                .padding()
+            Spacer()
```
themeButtonView(), Spacer()가 중복되네요!
나중에 .fashion이 추가되는 등 주제가 변경, 추가, 삭제가 이뤄질때 코드를 한곳만 수정해도 짠 나올 수 있게 
좀 더 편리하게 짜는 방법이 혹시 있을까요? 🤔

kuk6933 replied:
중복되는 코드를 지우고 ThemeButton, PlusMinusButton파일에서 Spcaer()과 padding 추가했습니다!

kuk6933 replied:
리뷰 감사합니다!

rose6649 commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+    case fruit
+    var emojis: [String] {
+        switch self {
+        case .emotion:
+            return ["🥺", "🥸", "🤩", "😜", "🧐", "😤", "😛", "🥳", "🥶", "😓", "🫡", "🫥", "😝", "🤑", "😎", "🤓", "🤠",  "🤡", "😏", "😶", "😐", "🫤", "😑", "😒", "🙄", "🤨", "🤔", "🤫"]
+        case .animal:
+            return ["🐶", "🐱","🐭", "🐰", "🐻", "🐼", "🐨", "🦁", "🐮", "🐷", "🐥", "🦊", "🐵", "🦄", "🦓" ]
+        case .fruit:
+            return ["🍏", "🍎", "🍐", "🍊", "🍋", "🍌", "🍉", "🍇", "🍓", "🍈"]
```
이왕이면 case 순서와 맞춰주면 더 좋을 것 같아요~!
animal, emotion, fruit

enum 사용 👍

kuk6933 replied:
넵 순서 수정했습니다!

rose6649 commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+            return ["🍏", "🍎", "🍐", "🍊", "🍋", "🍌", "🍉", "🍇", "🍓", "🍈"]
+        }
+    }
+    var symbols: Image {
+        switch self {
+        case .animal:
+            return Image(systemName: "pawprint.circle")
+        case .emotion:
+            return Image(systemName: "face.smiling")
+        case .fruit:
+            return Image(systemName: "applelogo")
```
Image는 하나의 개체만 반환하는데 변수 이름은 symbols네요! ☺️

kuk6933 replied:
세개의 테마를 반환할수 있어서 복수형을 사용했었던것 같습니다. 수정했습니다!

rose6649 commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+        }
+    }
+    
+    var maxNumber: Int {
+        switch self {
+        case .animal:
+            return self.emojis.count
+        case .emotion:
+            return self.emojis.count
+        case .fruit:
+            return self.emojis.count
+        }
+    }
```
다 똑같은 것을 반환하지 않나용?

kuk6933 replied:
엇 다같은..코드네요 수정했습니다!

rose6649 reviewed (commented):
수고하셨어요 형석뉨~~!!!
첫 과제라 힘드셨을텐데 테마설정도 enum으로 깔끔하게 작성해주셨네요
다음 과제도 화이팅입니다!!! 👍👍🔥

leejun6694 commented on week1/Memorize/Memorize/CalculateButtonView.swift:
```diff
@@ -0,0 +1,45 @@
+    }
+    
+    var body: some View {
+        if(isPlus)
```
swift 에선 `if 조건`에서 `괄호`를 보통 생략합니다.

leejun6694 replied:
`중괄호`를 줄바꿈하는 곳과 안하는 곳이 있는데, 하나의 규칙으로 통일해주세요.

kuk6933 replied:
넵 괄호 생략하고 통일에 유의하겠습니다!

kuk6933 replied:
리뷰 감사합니다!

leejun6694 commented on week1/Memorize/Memorize/CalculateButtonView.swift:
```diff
@@ -0,0 +1,45 @@
+                }
+            } label: {
+                Image(systemName: "plus.circle")
+                    .font(.system(size: 20))
```
size 의 크기는 `CGFloat` 지만, 우리는 간편하게 `Int` 형으로 작성할 수 있습니다.
내부적으론 어떤 로직이 진행될까요?
한번 고민해봐요~

kuk6933 replied:
Int -> CGFloat로 형변환이 일어날것이라고 생각합니다!

leejun6694 commented on week1/Memorize/Memorize/ContentView.swift:
```diff
@@ -0,0 +1,36 @@
+import SwiftUI
+
+struct ContentView: View {
+
```
`줄바꿈 규칙`이 다른 곳과는 좀 다르네요~?

kuk6933 replied:
규칙 세워서 통일하겠습니다!

leejun6694 commented on week1/Memorize/Memorize/CardView.swift:
```diff
@@ -0,0 +1,38 @@
+    }
+}
+
+
```
공백

kuk6933 replied:
수정했습니다!

leejun6694 commented on week1/Memorize/Memorize/CardView.swift:
```diff
@@ -0,0 +1,38 @@
+
+struct CardView_Previews: PreviewProvider {
+    static var previews: some View {
+        CardView( content: "abc")
```
공백?

kuk6933 replied:
수정했습니다!

leejun6694 commented on week1/Memorize/Memorize/Model/Theme.swift:
```diff
@@ -0,0 +1,47 @@
+enum Theme: Comparable{
+    case animal 
+    case emotion
+    case fruit
```
`case` 아래로 줄바꿈이 하나 필요할거 같네요?

kuk6933 replied:
가독성을 위해 case와 다른 변수들을 공백으로 분리하는게 필요하다는 의미인지 여쭤보고 싶습니다! 
수정했습니다!

leejun6694 commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+    func returnCardList() -> [String]
+    {
+        var cardList: [String] = []
+        for (idx, emoji) in themeEmojis.enumerated() {
```
가능한 축약형을 쓰지 않습니다. 네이밍이 길어지더라도 의미가 온전히 전달될 수 있도록 풀네임을 써주세요
idx -> index

kuk6933 replied:
넵 풀네임을 사용하겠습니다

leejun6694 commented on week1/Memorize/Memorize/Model/ToBeObserved.swift:
```diff
@@ -0,0 +1,37 @@
+        }
+        return cardList
+    }
+    
```
공백

kuk6933 replied:
수정했습니다!

kuk6933 committed:
Update to reflect a review
Details
- code convention
- delete redundancy
- delete unnecessary code
- make Model/PlusOrMinus.swift file

kuk6933 committed:
Naming update
-ButtonView -> ButtonBar
-toBeObserved -> Admin //difficult

kuk6933 committed:
Naming Update

gen-com commented on week1/Memorize/Memorize/ContentView.swift:
```diff
@@ -0,0 +1,28 @@
+                    .font(.largeTitle)
+                    .foregroundColor(.black)
+                CardListView()
+                    .environmentObject(memorizeGame)
```
여기랑

gen-com commented on week1/Memorize/Memorize/ContentView.swift:
```diff
@@ -0,0 +1,28 @@
+            }
+            Spacer()
+            ButtonBar()
+                .environmentObject(memorizeGame)
```
여기 따로 추가한 이유는 뭔가요 ?

environmentObject는 모든 뷰라기 보다 현재 뷰와 그 하위뷰들에 대해 적용되는게 아닌가요 ?

kuk6933 replied:
저도 그렇게 생각해서 처음에 적지않았었는데
No ObservableObject of type MemorizeGame found. A View.environmentObject(_:) for MemorizeGame may be missing as an ancestor of this view. 이런 오류가 떠서 구글링한 결과 이 코드들을 추가해서 해결했었습니다.

kuk6933 replied:
MemorizeApp에서 ContentView에 넘겨주는것으로 수정했습니다

kuk6933 committed:
Update for EnvironmentObject
MemorizeApp에서 MemorizeGame인스턴스를 만들고 ContentView에 pass해서 한번만 넘겨도 되도록 했습니다.

kuk6933 committed:
Edit Model
delete import SwiftUI
change symbol type Image -> String

kuk6933 committed:
alter MemorizeGame location

kuk6933 committed:
last edit

leejun6694 commented:
고생했어요~!
다만, commit 메세지를 좀 더 고민해볼까요?
last edit 이런 메세지가 추후에 이 commit 은 어떤 수정이었는지 유추하기 힘들겠네요
